### PR TITLE
WIP: Add method to ApiClient to load all entries when results are paginated

### PR DESCRIPTION
**Pull request by thathoff**
The library currently does not support paginated results and the Tick API returns only the first 100 entries. I’ve implemented a simple getAll() to load all results of possible paginated request.

If this looks reasonable to you I would replace all ApiClient::get() calls with ApiClient::getAll() where paginated results are expected.

### CODE REVIEW

**Review by jltorresm — COMMENTED** [review]
Tickspot API gives you implicit control over the paginated results with the requirement of using `start_date` & `end_date` for the entry endpoint. 

There could be the case that even when restricting to a time span you have more than a hundred tasks (when getting a complete year worth of entries for example). But for that particular case, I'd rather give the developer a way of asking for next pages intentionally, than invisibly making multiple external requests and aggregating internally the full result set. 

The _why_ becomes apparent in a case where, for some reason, you need to get the full history of a particular user. Instead of getting a huge array of hundreds or thousands of entries and then, potentially, applying again some kind of pagination on the presentation layer. You could just take advantage of the existing pagination, and show results as you go. 

So in summary, it definitely needs updates to provide access to the full result set. But preferably taking advantage of the existing Tickspot API's pagination, forwarding this ability to the library user. If you want to propose a different approach aligned with this comment, make a new PR and I'll be happy to check it as soon as possible.

**Comment by thathoff** [comment]
Ok, seems reasonable! :) What about a parameter to methods like Task::getList() to set the page. So one can simply call `$tick->task->getList(2)` to get the second page of results?

**Comment by jltorresm** [comment]
Yes, that sounds good. We would need to add that to `Task::getList()`, `Project::getList()` and `Entry::getList(..., $page)`
